### PR TITLE
Replace MAINTAINERS with CODEOWNERS

### PR DESCRIPTION
**Pull request by penberg**
Replace the MAINTAINERS file with a CODEOWNERS file, which Github is
able to parse, and suggest reviewers for pull requests.

### CODE REVIEW

File: scripts/find-maintainer [hunk 1]
```diff
@@ -1,89 +0,0 @@
-#!/usr/bin/env python3
-# -*- coding: utf-8 -*-
-
```
Comment (avikivity) [inline scripts/find-maintainer:3]: Did you delete this file on purpose?

Comment (penberg) [reply scripts/find-maintainer:3]: @avikivity Yes, once `MAINTAINERS` is removed, this script no longer makes sense.

Comment (denesb) [reply scripts/find-maintainer:3]: Can't the script just use the new file? Not everybody uses github, although I'm not sure anybody ever used this script at all.

Comment (psarna) [reply scripts/find-maintainer:3]: @denesb I did, once. Just to realize it was not up-to-date though :)

Comment (penberg) [reply scripts/find-maintainer:3]: @denesb There's a cost in porting `find-maintainers` to use the new format, and I am less than convinced it's worth it. For starters, I am not aware of anyone ever actually using the script. Second, on "not everybody uses Github" -- that ship has sailed a long time ago (maybe we left a few people stranded on the island of emails, but...). Third, there are existing tools out there that parse `CODEOWNERS`. See for example https://github.com/nmusolino/codeowners. Anyone that needs to parse this information outside GitHub can use one of them.

Comment (denesb) [reply scripts/find-maintainer:3]: @penberg I'm not insisting we do this porting, I was just curious why you decided it "no longer makes sense". I agree we shouldn't waste time on keeping this script alive at all cost.

Comment (penberg) [reply scripts/find-maintainer:3]: Merge ping!

**Pull request merged**